cas-server-support-x509/src/main/java/org/jasig/cas/adaptors/x509/authentication/handler/support/ldap/LdaptiveResourceCRLFetcher.java

```diff
@@ -50,22 +56,59 @@ public LdaptiveResourceCRLFetcher(
         this.searchExecutor = searchExecutor;
     }
 
+    private boolean isLdap(final String r) {
+        return r.toLowerCase().startsWith("ldap");
+    }
+
+    private boolean isLdap(final URI r) {
+        return r.getScheme().equalsIgnoreCase("ldap");
+    }
+
+    private boolean isLdap(final URL r) {
+        return r.getProtocol().equalsIgnoreCase("ldap");
```

Comment:
Gotcha. I meant that you could probably do something like this:

```
return isLdap(r.getScheme());
...
return isLdap(r.getProtocol());
```

This should work.